src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/RenameTypeService.cs

```diff
@@ -0,0 +1,239 @@
+                return false;
+
+            if (!TryGetCodeModel(newFilePath, project, out EnvDTE.FileCodeModel codeModel))
+                return false;
+
+            (bool success, bool isCaseSensitive) = await TryDetermineIfCompilationIsCaseSensitiveAsync(project);
+            if (!success)
+                return false;
+
+            if (!TryGetCodeElementToRename(oldName, codeModel.CodeElements, isCaseSensitive, out EnvDTE80.CodeElement2 codeElementToRename))
+                return false;
+
+            codeElementToRename.RenameSymbol(newName);
+            return true;
+        }
+
+        private bool TryGetCodeModel(string newFilePath, Project project, out EnvDTE.FileCodeModel codeModel)
+        {
+            codeModel = null;
+            Document newDocument = GetDocument(project, newFilePath);
+            if (newDocument is null)
+                return false;
+
+            IVsHierarchy hierarchy = _workspace.GetHierarchy(project.Id);
+            if (hierarchy is null)
+                return false;
+
+            if (ErrorHandler.Failed(hierarchy.GetProperty(VSConstants.VSITEMID_ROOT, (int)__VSHPROPID.VSHPROPID_ExtObject, out object value)))
+                return false;
+
+            var dteProject = (EnvDTE.Project)value;
+
+            EnvDTE.ProjectItem projectItemForDocument = dteProject.ProjectItems
+                .OfType<EnvDTE.ProjectItem>()
+                .FirstOrDefault(p => StringComparer.InvariantCultureIgnoreCase.Compare(p.Name, newDocument.Name) == 0);
+
+            codeModel = projectItemForDocument.FileCodeModel;
+            return true;
+        }
+
+        private bool TryGetProjectAtPath(string fullPath, out Project project)
+        {
+            // NOTE: It is safe to grab the first project we encounter which
+            // contains the given file. Roslyn will handle the case where the
+            // file is included in multiple projects (linked-files or multi-TFM)
+            foreach (Project proj in _workspace.CurrentSolution.Projects)
+            {
+                if (StringComparers.Paths.Equals(proj.FilePath, fullPath))
+                {
+                    project = proj;
+                    return true;
+                }
+
+            }
+
+            project = null;
+            return false;
+        }
+
+        private async Task<bool> CanHandleRenameAsync(string oldName, string newName, Project project)
+        {
+            (bool success, bool isCaseSensitive) = await TryDetermineIfCompilationIsCaseSensitiveAsync(project);
+            if (!success)
+                return false;
+
+            return IsValidIdentifier(oldName) &&
+                   IsValidIdentifier(newName) &&
+                       (!string.Equals(
+                            oldName,
+                            newName,
+                            isCaseSensitive
+                                ? StringComparison.Ordinal
```

Comment:
Why not just expose the right string comparer from the ISyntaxFactsService? That's what the Roslyn ISyntaxFactsService exposes; not sure why you wouldn't do the same on your copy rather than all the switching here.